pkg/controller/datavolume/pvc-clone-controller.go

```diff
@@ -278,9 +278,9 @@ func (r *PvcCloneReconciler) syncClone(log logr.Logger, req reconcile.Request) (
 			return syncRes, nil
 		}
 
-		// Always call detect size, it will handle the case where size is specified
-		// and detection pod not necessary
-		if datavolume.Spec.Storage != nil {
+		// If the target's size is not specified, we can extract that value from the source PVC
+		targetRequest, hasTargetRequest := pvcSpec.Resources.Requests[corev1.ResourceStorage]
+		if !hasTargetRequest || targetRequest.IsZero() {
```

Comment:
Looks good, but maybe add a utest for it? also the func sure needs to be refactored... so many tabs I get a headache :)